packages/synapse-react-client/src/synapse-client/SynapseClient.test.ts

```diff
@@ -0,0 +1,133 @@
+import SynapseClient, { FunctionReturningPaginatedResults } from './index'
+import { SynapseClientError } from '../utils/SynapseClientError'
+import {
+  ACCESS_TOKEN_COOKIE_KEY,
+  NETWORK_UNAVAILABLE_MESSAGE,
+} from '../utils/SynapseConstants'
+import { PaginatedResults } from '@sage-bionetworks/synapse-types'
+import UniversalCookies from 'universal-cookie'
+import * as SynapseClientUtils from './SynapseClientUtils'
+import * as HttpClient from './HttpClient'
+
+const isOutsideSynapseOrgSpy = jest.spyOn(
+  SynapseClientUtils,
+  'isOutsideSynapseOrg',
+)
+
+const doGetSpy = jest.spyOn(HttpClient, 'doGet')
+
+describe('SynapseClient tests', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+  describe('getAccessTokenFromCookie', () => {
+    it('No cookie outside of *.synapse.org', async () => {
+      isOutsideSynapseOrgSpy.mockReturnValue(true)
+      new UniversalCookies().remove(ACCESS_TOKEN_COOKIE_KEY, { path: '/' })
+      window.location.assign('http://localhost:3000/')
+      expect(await SynapseClient.getAccessTokenFromCookie()).toBeUndefined()
+    })
+    it('Has cookie outside of *.synapse.org', async () => {
+      isOutsideSynapseOrgSpy.mockReturnValue(true)
+      const token = 'fake-token'
+      new UniversalCookies().set(ACCESS_TOKEN_COOKIE_KEY, token, {
+        path: '/',
+      })
+      window.location.assign('http://localhost:3000/')
+      expect(await SynapseClient.getAccessTokenFromCookie()).toBe(token)
+    })
+    it('Calls portal servlet to get token', async () => {
+      isOutsideSynapseOrgSpy.mockReturnValue(false)
+      const token = 'fake-token'
+      window.location.assign('http://somesubdomain.synapse.org/')
+      doGetSpy.mockResolvedValue(token)
+      expect(await SynapseClient.getAccessTokenFromCookie()).toBe(token)
+
+      expect(doGetSpy).toHaveBeenCalled()
+    })
+    it('Calls portal servlet and is unauthenticated', async () => {
+      isOutsideSynapseOrgSpy.mockReturnValue(false)
+      window.location.assign('http://somesubdomain.synapse.org/')
+      // Calling the sessioncookie service without a cookie set will result in a 401
+      doGetSpy.mockRejectedValue(
+        new SynapseClientError(
+          401,
+          'token unavailable',
+          expect.getState().currentTestName!,
+        ),
+      )
+      await expect(() =>
+        SynapseClient.getAccessTokenFromCookie(),
+      ).rejects.toThrow()
+
+      expect(doGetSpy).toHaveBeenCalled()
+    })
+  })
```

Comment:
Added these because the type was wrong and I wanted to figure out why.

The function claimed to return `string | null` but actually returns `string | undefined` and throws an error when hitting the servlet. We could/should better handle that case, but I didn't want to change it in this PR.